### DISCUSSION TITLE
Contributing / Contracts

### DISCUSSION THREAD

**Discussion by crypticcrow42** [pwillworth/dfkreport#5] [General]
I was interested in contributing and cloned the repo / got it working via command line.  I was never able to get the web portion fully running on a xampp server with cgi enabled / maria db setup correctly, it would just spin on generate with no errors.  How would you suggest setting up a development environment?

I have a friend who works with defi on accounting and dfkreport gets almost everything he needs but theres a few things he'd like.  I could work on these once I get a development environment up, he also mentioned potentially getting contracts for his requests if you are interested in looking at them for compensation?  

Thanks, if you want to respond personally my email is crypticcrow42@gmail.com

**Comment by pwillworth** [comment]
Hi, that's great!  Happy to see some contributions for sure.

For you development environment, probably all you need to do is kick off the watcher process.  So, just go to your web directory in a terminal and run ../watch.py  This process monitors the db for report requests that are put there by generate.py and spawns a separate process that does the work.

**Reply by crypticcrow42** [reply]
Thanks!  Got it working with that!

**Reply by crypticcrow42** [reply]
Would it be possible to do a google meet sometime with me, my accounting friend RoosterBigz, FriskyFox, and maybe a couple others from the defi team? I believe it would be about expanding / adding features to dfkreport that would be useful for him, and the defi community as well, with a potential grant to do the work?

**Reply by pwillworth** [reply]
Yeah, I don't think we could get the DFK team on a call, that's not really how the grant program works.  The three of us can get together, talk about your ideas, and put a grant proposal together though.  I'd be down with that.  Will follow up on email to plan.
